src/Coq.hs

```diff
@@ -158,30 +162,43 @@ stateval
   -> (Id -> SlotType -> T.Text)
   -> [StorageUpdate]
   -> T.Text
-stateval store handler updates = T.unwords $ stateConstructor : fmap (valuefor updates) (M.toList store)
-  where
-  valuefor :: [StorageUpdate] -> (Id, SlotType) -> T.Text
-  valuefor updates' (name, t) =
-    case find (eqName name) updates' of
-      Nothing -> parens $ handler name t
-      Just (Update SByteStr _ _) -> error "bytestrings not supported"
-      Just (Update _ item e) -> lambda (ixsFromItem item) 0 e (idFromItem item)
+stateval store handler updates = T.unwords $ stateConstructor : fmap (updateVar updates handler) (M.toList store)
 
 -- | filter by name
 eqName :: Id -> StorageUpdate -> Bool
 eqName n update = n == idFromUpdate update
 
--- represent mapping update with anonymous function
-lambda :: [TypedExp] -> Int -> Exp a -> Id -> T.Text
-lambda [] _ e _ = parens $ coqexp e
-lambda (TExp argType arg:xs) n e m = parens $
-  "fun " <> name <> " =>"
-  <> " if " <> name <> eqsym <> coqexp arg
-  <> " then " <> lambda xs (n + 1) e m
-  <> " else " <> T.pack m <> " " <> stateVar <> " " <> lambdaArgs n where
-  name = anon <> T.pack (show n)
-  lambdaArgs i = T.unwords $ map (\a -> anon <> T.pack (show a)) [0..i]
-  eqsym = case argType of
+-- | Returns the updated value of a storage variable 
+updateVar :: [StorageUpdate]
+          -> (Id -> SlotType -> T.Text)
+          -> (Id, SlotType)
+          -> T.Text
+updateVar updates handler (name, t@(StorageValue _)) = 
+  parens $ foldl updatedVal (handler name t) (filter (eqName name) updates) where
+
+  updatedVal _ (Update SByteStr _ _) = error "bytestrings not supported"
+  updatedVal _ (Update _ _ e) = coqexp e
```

Comment:
Just a v small stylistic nit, but I prefer to keep the where on the line below, smth like this:

```suggestion
  parens $ foldl updatedVal (handler name t) (filter (eqName name) updates) 
    where
      updatedVal _ (Update SByteStr _ _) = error "bytestrings not supported"
      updatedVal _ (Update _ _ e) = coqexp e
```

Reply:
Fixed